verification/src/compiler/fetcher.rs

```diff
@@ -6,4 +6,5 @@ use std::path::PathBuf;
 pub trait Fetcher {
     type Error;
     async fn fetch(&self, ver: &CompilerVersion) -> Result<PathBuf, Self::Error>;
+    fn all_versions(&self) -> Vec<&CompilerVersion>;
```

Comment:
I think it would be better to split the trait into two `Fetcher` and `VersionList` or something

Reply:
so in the tests, for example, you wouldn't need to impl empty function for mock fetchers